src/engraving/libmscore/cmd.cpp

```diff
@@ -777,15 +777,19 @@ Segment* Score::setNoteRest(Segment* segment, int track, NoteVal nval, Fraction
     Element* nr   = nullptr;
     Tie* tie      = nullptr;
     ChordRest* cr = toChordRest(segment->element(track));
-
+    Tuplet* tuplet = cr ? cr->tuplet() : 0;
     Measure* measure = nullptr;
+    bool targetIsRest = cr && cr->isRest();
     for (;;) {
         if (track % VOICES) {
             expandVoice(segment, track);
         }
-
+        if (targetIsRest && !cr->isRest()) {
+            undoRemoveElement(cr);
+            segment = addRest(segment, track, cr->ticks(), cr->tuplet())->segment();
+        }
         // the returned gap ends at the measure boundary or at tuplet end
-        Fraction dd = makeGap(segment, track, sd, cr ? cr->tuplet() : 0);
+        Fraction dd = makeGap(segment, track, sd, tuplet);
```

Comment:
the changes here are about ensuring that a) if the element at the segment position is a rest, then any other notes that would be overlapped get removed instead of being "split" into component durations - actually I'd rather simplify this so that setNoteRest *always* does that but it may have unexpected side effects and b) only the *initial* tuplet is preserved where possible - the existing logic tried to preserve all tuplets that fell inside the duration, which didn't make sense for any case I could think of